data_override/data_override.F90

```diff
@@ -692,610 +230,53 @@ subroutine data_override_3d(gridname,fieldname_code,data,time,override,data_inde
-  integer, intent(in), optional :: data_index
-
-  character(len=512) :: filename !< file containing source data
-  character(len=128) :: fieldname !< fieldname used in the data file
-  integer :: index1 !< field index in data_table
-  integer :: id_time !< index for time interp in override array
-  integer :: curr_position !< position of the field currently processed in override_array
-  integer :: i
-  real :: factor
-
-  if(.not.module_is_initialized) &
-       call mpp_error(FATAL,'Error: need to call data_override_init first')
-
-!1  Look  for the data file in data_table
-  if(PRESENT(override)) override = .false.
-  if (present(data_index)) then
-    index1 = data_index
-  else
-    index1 = -1
-    do i = 1, table_size
-       if( trim(gridname) /= trim(data_table(i)%gridname)) cycle
-       if( trim(fieldname_code) /= trim(data_table(i)%fieldname_code)) cycle
-       index1 = i                               ! field found
-       exit
-    enddo
-    if(index1 .eq. -1) then
-       if(debug_data_override) &
-         call mpp_error(WARNING,'this field is NOT found in data_table: '//trim(fieldname_code))
-       return  ! NO override was performed
-    endif
-  endif
-
-  fieldname = data_table(index1)%fieldname_file ! fieldname in netCDF data file
-  factor = data_table(index1)%factor
-
-  if(fieldname == "") then
-     data = factor
-     if(PRESENT(override)) override = .true.
-     return
-  else
-     filename = data_table(index1)%file_name
-     if (filename == "") call mpp_error(FATAL,'data_override: filename not given in data_table')
-  endif
-
-!3 Check if fieldname has been previously processed
-!$OMP SINGLE
-  curr_position = -1
-  if(num_fields > 0 ) then
-     do i = 1, num_fields
-        if(trim(override_array(i)%gridname) /= trim(gridname))   cycle
-        if(trim(override_array(i)%fieldname) /= trim(fieldname_code)) cycle
-        curr_position = i
-        exit
-     enddo
-  endif
-
-  if(curr_position < 0) then ! the field has not been processed previously
-     num_fields = num_fields + 1
-     curr_position = num_fields
-     ! record fieldname, gridname in override_array
-     override_array(curr_position)%fieldname = fieldname_code
-     override_array(curr_position)%gridname = gridname
-     id_time = init_external_field(filename,fieldname,verbose=.false.)
-     if(id_time<0) call mpp_error(FATAL,'data_override:field not found in init_external_field 1')
-     override_array(curr_position)%t_index = id_time
-  else !curr_position >0
-     !9 Get id_time  previously stored in override_array
-     id_time = override_array(curr_position)%t_index
-  endif !if curr_position < 0
-
-  !10 do time interp to get data in compute_domain
-  call time_interp_external(id_time, time, data, verbose=.false.)
-  data = data*factor
-!$OMP END SINGLE
-
-  if(PRESENT(override)) override = .true.
-
-end subroutine data_override_0d
-
-!> @brief Data override for 2D unstructured grids
+!> @brief Data override for 1D unstructured grids
```

Comment:
This needs to be updated to reflect what the routine is actually doing

Reply:
This subroutine was removed in b89c8db621164eef22e0d9220a2046a4a1ceaa98.